include/bout/mesh.hxx

```diff
@@ -514,6 +552,8 @@ class Mesh {
   //////////////////////////////////////////////////////////
   
   int GlobalNx, GlobalNy, GlobalNz; ///< Size of the global arrays. Note: can have holes
+  /// Size of the global arrays excluding boundary points.
+  int GlobalNxNoBoundaries, GlobalNyNoBoundaries, GlobalNzNoBoundaries;
```

Comment:
Some way to get `GlobalNyNoBoundaries` is very useful, because the number of y-boundary cells can change depending on whether there is a second X-point or not.

The `*NoBoundaries` versions correspond to the `getGlobal*IndexNoBoundaries()` methods, so are useful for things like "am I within 8 grid points of the outer radial boundary?".

Is there a different interface for getting this kind of data that you'd rather have? If we could access `GlobalNx`, etc. through there we could also remove them from the public interface.
* Maybe something like `gnx = mesh->get("GlobalNx");`, except we need a different name because `Mesh` already has a `get` method? I guess this would be implemented with a `std::map` that we could initialise at the end of `BoutMesh::load()` - is the cost of a lookup in a small `map` worth worrying about at all (assuming we don't call this method inside tight loops)?

Reply:
I think just wrapping them in a getter and not providing a setter:

```cpp
class Mesh {
private:
  int global_nx_no_boundaries;
public:
  int Mesh::getGlobalNxNoBoundaries() { return global_nx_no_boundaries; }
};
```

There's a lot of stuff that are currently just public data members that should not be, for example `LocalN*`, `GlobalN*`, all the metric components in `Coordinates`, and so on.

It could be a lot of pain to convert everything to use getters, but it would be more correct.

Leave them as public members for now, otherwise it would be weirdly inconsistent, which is probably worse!

Reply:
Just to note, the metric components should probably stay public. It's likely that users (e.g. STORM) want to normalise lengths, which means the metric components need to be rescaled in physics-model code.

Reply:
Oh, for sure those also want a setter, but it should be a function so that we can then recalculate things that need to be consistent